src/actions/mod.rs

```diff
@@ -35,6 +35,21 @@ use std::time::Duration;
 
 use self::compiler_message_parsing::{FileDiagnostic, ParseError, Suggestion};
 
+// TODO: Support non-`file` URI schemes in VFS. We're currently ignoring them because
+// we don't want to crash the RLS in case a client opens a file under different URI scheme
+// like with git:/ or perforce:/ (Probably even http:/? We currently don't support remote schemes).
+macro_rules! ignore_non_file_uri {
```

Comment:
I would probably move the call to `parse_file_path` into this macro. However, what to do about spans? Maybe call this macro inside the span parsing code and propagate an error later? If that is too much work we could have a macro that parses a uri string to a path or does the error handling like this and another than just does this.